src/morphodict/lexicon/importer/test_importer.py

```diff
@@ -0,0 +1,530 @@
+
+
+@parametrize_incremental
+@parametrize_translate_wordforms
+def test_non_lemma_wordform_added(db, translate_wordforms, incremental):
+    import_test_file(
+        "single-word.importjson",
+        translate_wordforms=translate_wordforms,
+        incremental=incremental,
+    )
+
+    assert Wordform.objects.filter(is_lemma=True).count() == 1
+    assert Wordform.objects.get(slug="maskwa")
+
+    if translate_wordforms:
+        assert (
+            Wordform.objects.get(text="maskonâhk")
+            .definitions.filter(auto_translation_source_id__isnull=True)
+            .count()
+            == 0
+        )
+    else:
+        assert Wordform.objects.count() == 1
+
+    import_test_file(
+        "single-word-plus-wordform.importjson",
+        translate_wordforms=translate_wordforms,
+        incremental=incremental,
+    )
+
+    wf = Wordform.objects.get(text="maskonâhk")
+    assert definitions_match(
+        wf.definitions.filter(auto_translation_source__isnull=True),
+        [["among the bears", "X"]],
+    )
+
+
+@parametrize_incremental
+@parametrize_translate_wordforms
+def test_non_lemma_wordform_removed(db, translate_wordforms, incremental):
+    import_test_file(
+        "single-word-plus-wordform.importjson",
+        translate_wordforms=translate_wordforms,
+        incremental=incremental,
+    )
+
+    wf = Wordform.objects.get(text="maskonâhk")
+    assert definitions_match(
+        wf.definitions.filter(auto_translation_source__isnull=True),
+        [["among the bears", "X"]],
+    )
+
+    import_test_file(
+        "single-word.importjson",
+        translate_wordforms=translate_wordforms,
+        incremental=incremental,
+        purge=True,
+    )
+
+    assert Wordform.objects.filter(is_lemma=True).count() == 1
+    assert Wordform.objects.get(slug="maskwa")
+
+    if translate_wordforms:
+        assert (
+            Wordform.objects.get(text="maskonâhk")
+            .definitions.filter(auto_translation_source_id__isnull=True)
+            .count()
+            == 0
+        )
+    else:
+        assert Wordform.objects.count() == 1
+
+
+@parametrize_translate_wordforms
+def test_non_lemma_definition(db, translate_wordforms):
+    assert Wordform.objects.count() == 0
+
+    import_test_file(
+        "non-lemma-definition.importjson", translate_wordforms=translate_wordforms
+    )
+
+    lemma = Wordform.objects.get(slug="nîmiw")
+    non_lemma = Wordform.objects.get(text="nîminâniwan")
+
+    assert non_lemma.lemma == lemma
+
+    assert non_lemma.raw_analysis == [[], "nîmiw", ["+V", "+AI", "+Ind", "+X"]]
+
+    expected_defns = [["it is a dance, a time of dancing", "CW"]]
+    if translate_wordforms:
+        expected_defns.append(["someone dances", "🤖CW"])
+    assert definitions_match(non_lemma.definitions.all(), expected_defns)
+
+    assert {"time", "danc"}.issubset(
+        {t.text for t in TargetLanguageKeyword.objects.filter(wordform=non_lemma)}
+    )
```

Comment:
These tests are very thorough--I like it.

Reply:
😀